Goroutine may leak in test

## Enhancement

```
PASS
[2025/07/03 03:26:49.368 +00:00] [ERROR] [subscribe.go:170] ["Error processing events"] [category=ddl-notifier] [error="session pool closed"] [errorVerbose="session pool closed\ngithub.com/pingcap/errors.Trace\n\texternal/com_github_pingcap_errors/juju_adaptor.go:15\ngithub.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).processEvents\n\tpkg/ddl/notifier/subscribe.go:183\ngithub.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start\n\tpkg/ddl/notifier/subscribe.go:163\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1\n\tpkg/util/wait_group_wrapper.go:189\nruntime.goexit\n\tsrc/runtime/asm_amd64.s:1700"] [stack="github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start\n\tpkg/ddl/notifier/subscribe.go:170\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1\n\tpkg/util/wait_group_wrapper.go:189"]
coverage: 9.3% of statements
[2025/07/03 03:26:50.368 +00:00] [ERROR] [subscribe.go:170] ["Error processing events"] [category=ddl-notifier] [error="session pool closed"] [errorVerbose="session pool closed\ngithub.com/pingcap/errors.Trace\n\texternal/com_github_pingcap_errors/juju_adaptor.go:15\ngithub.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).processEvents\n\tpkg/ddl/notifier/subscribe.go:183\ngithub.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start\n\tpkg/ddl/notifier/subscribe.go:163\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1\n\tpkg/util/wait_group_wrapper.go:189\nruntime.goexit\n\tsrc/runtime/asm_amd64.s:1700"] [stack="github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start\n\tpkg/ddl/notifier/subscribe.go:170\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1\n\tpkg/util/wait_group_wrapper.go:189"]
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 28119 in state select, with github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start on top of the stack:
github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start(0xc00964ecc0)
	pkg/ddl/notifier/subscribe.go:159 +0x1e5
github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1()
	pkg/util/wait_group_wrapper.go:189 +0x6b
created by github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover in goroutine 28460
	pkg/util/wait_group_wrapper.go:181 +0x99
 Goroutine 28460 in state select, with github.com/pingcap/tidb/pkg/util/timeutil.Sleep on top of the stack:
github.com/pingcap/tidb/pkg/util/timeutil.Sleep({0x73c7c78, 0xc009e686e0}, 0xc0068f9ee0?)
	pkg/util/timeutil/time.go:28 +0xcb
github.com/pingcap/tidb/pkg/owner.(*mockManager).CampaignOwner.func1()
	pkg/owner/mock.go:159 +0x345
created by github.com/pingcap/tidb/pkg/owner.(*mockManager).CampaignOwner in goroutine 131
	pkg/owner/mock.go:138 +0x6c
]
```

Sometimes the stats owner manager is not closed when the test is complete. It may fail the leak detection.

--- a/pkg/domain/domain.go
+++ b/pkg/domain/domain.go
@@ -2078,6 +2078,13 @@ func (do *Domain) gcStatsWorkerExitPreprocessing() {
 		do.statsOwner.Close()
 		ch <- struct{}{}
 	}()
+	if intest.InTest {
+		// We should wait the owner to be closed in test.
+		// Otherwise, the goroutine leak detection may fail.
+		<-ch
+		logutil.BgLogger().Info("gcStatsWorker exit preprocessing finished")
+		return
+	}
 	select {
 	case <-ch:
 		logutil.BgLogger().Info("gcStatsWorker exit preprocessing finished")

--- a/pkg/domain/domain.go
+++ b/pkg/domain/domain.go
@@ -2079,7 +2079,7 @@ func (do *Domain) gcStatsWorkerExitPreprocessing() {
 		ch <- struct{}{}
 	}()
 	if intest.InTest {
-		// We should wait the owner to be closed in test.
+		// We should wait for statistics owner to close on exit.
 		// Otherwise, the goroutine leak detection may fail.
 		<-ch
 		logutil.BgLogger().Info("gcStatsWorker exit preprocessing finished")

